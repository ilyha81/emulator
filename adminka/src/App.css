/* Чистим стили */
@import-normalize;

.App-Wrapper {
  display: grid;
  grid-template-areas: 'app';
  width: 100%;
  height: 100vh;
}

.App {
  grid-area: app;
  z-index: 1;
  display: grid;
  width: 100%;
  height: 100%;
  grid-template-rows: minmax(100px, 10vh) 1fr;
  grid-template-columns: minmax(400px, 20vw) 1fr;
  grid-template-areas: 'header header'
                        'sidebar main';
}

.Popup {
  grid-area: app;
  z-index: 10;
}





