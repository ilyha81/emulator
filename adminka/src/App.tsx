import React, {useEffect} from 'react';
import './App.css';
import {Header} from "./Menu/Header/Header";
import {Sidebar} from "./Menu/Sidebar/Sidebar";
import {Main} from "./Menu/Main/Main";
import {ESocketConnection, SocketAPI} from "./Socket/socket";
import {useSocketConnectionStatus} from "./Socket/SocketHooks";

function App () {

    /**
     * Socket Connection Status.
     */
    const connectionStatus:ESocketConnection = useSocketConnectionStatus();

    /**
     * Connecting to Server on start.
     * Subscribing on Messages and Connection Status.
     */
    useEffect(()=> {
        const subscription = SocketAPI.subscribeOnMessages((msg) => {
            if (msg.length > 0) {
                alert(msg)
            }
        });

        return () => {
            subscription.unsubscribe();
        }
    }, []);


    return (
        <div className="App-Wrapper">
            <div className="App">
                <Header socketStatus={connectionStatus} />
                <Sidebar socketStatus={connectionStatus}/>
                <Main/>
            </div>
            {/*<div className="Popup">

          </div>*/}
        </div>
    );
}

export default App;
