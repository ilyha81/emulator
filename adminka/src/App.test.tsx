import React from 'react';
import {render, screen} from '@testing-library/react';
import App from './App';

describe('Testing', () => {

    test('App render', () => {
        const {container} = render(<App/>);
        expect(container).toMatchSnapshot();
    });

    test('message connection', () => {
        // Надо замокать вызов SocketApi.connection и проверить вызовы на отрисовку и на анмаунт.
        const {container, unmount} = render(<App/>);

        unmount();

    });
});



