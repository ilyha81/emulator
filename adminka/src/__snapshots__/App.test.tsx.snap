// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render 1`] = `
<div>
  <div
    class="App-Wrapper"
  >
    <div
      class="App"
    >
      <div
        class="Header"
      >
        Хедер
        <div
          class="float-right"
          style="width: 160px; height: 100px; overflow: hidden;"
        >
          <img
            alt="HeaderLogo"
            class="logo "
            id="HeaderLogo"
            placeholder="HeaderLogo"
            src="logo.svg"
          />
          <div>
            <span
              class="badge badge-dark"
            >
              Socket: 
            </span>
            <span
              class="badge badge-warning"
            >
              INITIAL
            </span>
          </div>
        </div>
      </div>
      <div
        class="Sidebar container pt-2"
      >
        <button
          class="btn btn-block btn-lg btn-primary"
          disabled=""
        >
          Create Test Operation
        </button>
      </div>
      <div
        class="Main"
      >
        <div>
          Рабочий экран.
        </div>
      </div>
    </div>
  </div>
</div>
`;
